src/components/work/CancelBookingModal.tsx/CancelBookingModal.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import { ModalHeader, Modal, ModalBody, ModalFooter, Button } from 'reactstrap';
+import { GUEST_CANCEL_BOOKING, Booking, Currency, GET_GUEST_SORTED_BOOKINGS } from 'networking/bookings';
+import { graphql, compose } from 'react-apollo';
+import { differenceInDays } from 'date-fns';
+import { cancel, loadWeb3 } from 'utils/web3';
+import { AlertProperties } from 'components/work/Alert/Alert';
+import LoadingPortal from 'components/work/LoadingPortal';
+import { getFriendlyErrorMessage } from 'utils/validators';
+
+interface Props {
+  booking: Booking;
+  cancelBooking: (booking: Booking) => Promise<Booking>;
+  handleModalAction: () => void;
+  setAlert: (alert: AlertProperties) => void;
+}
+
+function CancelBookingModal({ booking, cancelBooking, handleModalAction, setAlert }: Props) {
+  const [isSubmitting, setSubmitting] = React.useState<boolean>(false);
+  const [currency, setCurrency] = React.useState<Currency | null>(Currency.USD);
+
+  if (isSubmitting) {
+    return <LoadingPortal currency={currency} />;
+  }
+
+  return (
+    <Modal isOpen toggle={() => !isSubmitting && handleModalAction}>
+      <ModalHeader>Cancel Booking</ModalHeader>
+      <ModalBody>
+        <h6>Are you sure you want to cancel this booking?</h6>
+        <h6>Booking: {booking.id}</h6>
+      </ModalBody>
+      <ModalFooter>
+        <Button color="secondary" disabled={isSubmitting} onClick={handleModalAction}>Back</Button>{' '}
+        <Button color="danger" disabled={isSubmitting} onClick={handleCancelBooking}>Yes, Cancel Booking</Button>
+      </ModalFooter>
+    </Modal>
+  );
+
+  function handleCancelBooking() {
+    if (!booking) return;
```

Comment:
This check is also unnecessary (handled at compile-time by enforcing Props interface)

Reply:
👍  the node backend in me must've gotten nervous =x